### PR TITLE
Add Cxx-Qt-Macro crate

### PR DESCRIPTION
**Pull request by LeonMatthesKDAB**
This allows us to add other items to cxx-qt, other than proc-macros. e.g. a QtObject trait or similar.

It might even allow us to move CxxQtThread into cxx-qt, instead of cxx-qt-lib.

### CODE REVIEW

File: crates/cxx-qt-macro/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+[package]
+name = "cxx-qt-macro"
+version.workspace = true
+authors = ["Andrew Hayzen <andrew.hayzen@kdab.com>","Leon Matthes <leon.matthes@kdab.com>", "Gerhard de Clercq <gerhard.declercq@kdab.com>"]
+edition.workspace = true
+license.workspace = true
+description = "A macro for Qt/C++ interop in Rust"
+homepage = "https://kdab.github.io/cxx-qt/book/"
+readme = "README.md"
+keywords = ["cxx", "ffi", "QML", "Qt"]
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-macro/Cargo.toml:16]: do our other sub crates we haven't been putting keywords, categories, homepage on them. Wonder if we should or shouldn't and just leave them for the main crate or have them everywhere? :thinking:

File: crates/cxx-qt-macro/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# SPDX-License-Identifier: MIT OR Apache-2.0
+[package]
+name = "cxx-qt-macro"
+version.workspace = true
+authors = ["Andrew Hayzen <andrew.hayzen@kdab.com>","Leon Matthes <leon.matthes@kdab.com>", "Gerhard de Clercq <gerhard.declercq@kdab.com>"]
+edition.workspace = true
+license.workspace = true
+description = "A macro for Qt/C++ interop in Rust"
+homepage = "https://kdab.github.io/cxx-qt/book/"
+readme = "README.md"
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-macro/Cargo.toml:15]: there is no readme in this crate, so probably remove like other crates?

File: crates/cxx-qt-macro/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+[lib]
+proc-macro = true
+
+[dependencies]
+cxx-qt-gen.workspace = true
+proc-macro2.workspace = true
+syn.workspace = true
+
+[dev-dependencies]
```
[the remark this answers is not in the document]
Comment (LeonMatthesKDAB) [reply crates/cxx-qt-macro/Cargo.toml:27]: Done

File: crates/cxx-qt-macro/Cargo.toml (hunk 1 above)
Comment (LeonMatthesKDAB) [reply crates/cxx-qt-macro/Cargo.toml]: Removed it

File: crates/cxx-qt-macro/Cargo.toml (hunk 2 above)
Comment (LeonMatthesKDAB) [reply crates/cxx-qt-macro/Cargo.toml]: Done

**Review by ahayzen-kdab — APPROVED** [review]
LGTM :-)

**Pull request merged**